geomesa-fs/geomesa-fs-datastore/src/main/scala/org/locationtech/geomesa/fs/FileSystemDataStore.scala

```diff
@@ -0,0 +1,107 @@
+  }
+
+  override def createSchema(sft: SimpleFeatureType): Unit = {
+    storage.createNewFeatureType(sft, PartitionScheme.extractFromSft(sft))
+  }
+
+}
+
+class FileSystemDataStoreFactory extends DataStoreFactorySpi {
+  import FileSystemDataStoreParams._
+  private val storageFactory = ServiceLoader.load(classOf[FileSystemStorageFactory])
+
+  override def createDataStore(params: util.Map[String, io.Serializable]): DataStore = {
+    import org.locationtech.geomesa.utils.conversions.ScalaImplicits.RichIterator
+    import scala.collection.JavaConversions._
+
+    val path = new Path(PathParam.lookUp(params).asInstanceOf[String])
+    val encoding = EncodingParam.lookUp(params).asInstanceOf[String]
+
+    val conf = new Configuration()
+    val storage = storageFactory.iterator().filter(_.canProcess(params)).map(_.build(params)).headOption.getOrElse {
+      throw new IllegalArgumentException("Can't create storage factory with the provided params")
+    }
+    val fs = path.getFileSystem(conf)
+
+    val readThreads = Option(ReadThreadsParam.lookUp(params)).map(_.asInstanceOf[java.lang.Integer])
+      .getOrElse(ReadThreadsParam.getDefaultValue.asInstanceOf[java.lang.Integer])
+
+    val ds = new FileSystemDataStore(fs, path, storage, readThreads, conf)
+    Option(NamespaceParam.lookUp(params).asInstanceOf[String]).foreach(ds.setNamespaceURI)
+    ds
+  }
+
+  override def createNewDataStore(params: util.Map[String, io.Serializable]): DataStore =
+    createDataStore(params)
+
+  override def isAvailable: Boolean = true
+
+  override def canProcess(params: util.Map[String, io.Serializable]): Boolean =
+    params.containsKey(PathParam.getName) && params.containsKey(EncodingParam.getName)
+
+  override def getParametersInfo: Array[DataAccessFactory.Param] =
+    Array(PathParam, EncodingParam, NamespaceParam)
+
+  override def getDisplayName: String = "File System (GeoMesa)"
+
+  override def getDescription: String = "File System Based Data Store"
+
+  override def getImplementationHints: util.Map[RenderingHints.Key, _] =
+    new util.HashMap[RenderingHints.Key, Serializable]()
+}
+
+object FileSystemDataStoreParams {
+  val PathParam            = new Param("fs.path", classOf[String], "Root of the filesystem hierarchy", true)
+  val EncodingParam        = new Param("fs.encoding", classOf[String], "Encoding of data", true)
+
+  val ConverterNameParam   = new Param("fs.options.converter.name", classOf[String], "Converter Name", false)
+  val ConverterConfigParam = new Param("fs.options.converter.conf", classOf[String], "Converter Typesafe Config", false)
+  val SftNameParam         = new Param("fs.options.sft.name", classOf[String], "SimpleFeatureType Name", false)
+  val SftConfigParam       = new Param("fs.options.sft.conf", classOf[String], "SimpleFeatureType Typesafe Config", false)
+
+  val ReadThreadsParam     = new Param("read-threads", classOf[java.lang.Integer], "Read Threads", false, 4)
```

Comment:
Is the 'read-threads' config particular to the fsds?  If so, should it be fs.read-threads?